lib/index.ts

```diff
@@ -1,28 +1,65 @@
 import 'source-map-support/register';
 import { getClassPathFromMvn } from './mvn-wrapper';
-import { getCallGraphGradle } from './gradle-wrapper';
+import { getClassPathFromGradle } from './gradle-wrapper';
 import { getCallGraph } from './java-wrapper';
 import { Graph } from '@snyk/graphlib';
 import { timeIt, getMetrics, Metrics } from './metrics';
+import {
+  CallGraphGenerationError,
+  ClassPathGenerationError,
+  EmptyClassPathError,
+  MissingTargetFolderError,
+  SubprocessTimeoutError,
+} from './errors';
 
 export async function getCallGraphMvn(
   targetPath: string,
   timeout?: number,
 ): Promise<Graph> {
-  const classPath = await timeIt('getMvnClassPath', () =>
-    getClassPathFromMvn(targetPath),
-  );
-  return await timeIt('getCallGraph', () =>
-    getCallGraph(classPath, targetPath, timeout),
-  );
+  try {
+    const classPath = await timeIt('getMvnClassPath', () =>
+      getClassPathFromMvn(targetPath),
+    );
+
+    return await timeIt('getCallGraph', () =>
+      getCallGraph(classPath, targetPath, timeout),
+    );
+  } catch (e) {
+    if (e instanceof MissingTargetFolderError) {
+      throw new CallGraphGenerationError(
+        "Could not find the project's target folder. Please compile your code by running `mvn compile` and try again.",
```

Comment:
Can we encapsulate those messages to be part of the error class? e.g.

```
export class MissingTargetFolderError extends Error {
  constructor(targetPath: string) {
    super(`Could not find the target folder starting in "${targetPath}"`);
    Object.setPrototypeOf(this, MissingTargetFolderError.prototype);
    this.userMessage = 'Could not find the project's target folder. Please compile your code by running `mvn compile` and try again.';
  }
}
```

And here it will be:
```
catch (e) {
  throw new CallGraphGenerationError(e.userMessage)
}
```